app/Console/Commands/FixTransaction.php

```diff
@@ -0,0 +1,199 @@
+
+            $aidstreamOrganizationIdentifierArray = $this->getAidstreamOrganizationIdentifier($aidstreamOrganizationIds);
+            $aidstreamOrganizationIdentifierArray = array_map('strtolower', $aidstreamOrganizationIdentifierArray);
+
+            $iatiOrganizations = Organization::whereIn('publisher_id', $aidstreamOrganizationIdentifierArray)->get();
+            $iatiOrganizationIdArray = $iatiOrganizations->pluck('id', 'publisher_id');
+
+            $idMap = $this->mapOrganizationIds($aidstreamOrganizationIdentifierArray, $iatiOrganizationIdArray);
+            $activities = Activity::whereIn('org_id', $idMap)->get();
+
+            foreach ($activities as $activity) {
+                $this->fixAllTransaction($activity);
+            }
+
+            $this->databaseManager->commit();
+            $this->info('Completed fixing transactions.');
+        } catch(\Exception $e) {
+            logger($e);
+        }
+
+        return 0;
+    }
+
+    /**
+     * Populate default values from activity and update transaction.
+     *
+     * @param $activity
+     *
+     * @return void
+     */
+    private function fixAllTransaction($activity): void
+    {
+        $defaultValues = json_encode([$activity->default_field_values]);
+        $transactions = $activity->transactions;
+
+        foreach ($transactions as $transaction) {
+            $tempTransaction = $transaction->transaction;
+            $tempTransaction = $this->populateDefaultFields($tempTransaction, $defaultValues);
+            $transaction->timestamps = false;
+            $transaction->updateQuietly(['transaction'=>$tempTransaction], ['touch'=>false]);
+        }
+    }
+
+    /**
+     * Ask input from user and return value.
+     *
+     * @param $question
+     * @param $field
+     * @param $rules
+     *
+     * @return string
+     */
+    protected function askValid($question, $field, $rules): string
+    {
+        $value = $this->ask($question);
+        $message = $this->validateInput($rules, $field, $value);
+
+        if ($message) {
+            $this->error($message);
+
+            return $this->askValid($question, $field, $rules);
+        }
+
+        return $value;
+    }
+
+    /**
+     * Validates input given by user.
+     *
+     * @param $rules
+     * @param $fieldName
+     * @param $value
+     *
+     * @return string|null
+     */
+    protected function validateInput($rules, $fieldName, $value): ?string
+    {
+        $validator = Validator::make([
+            $fieldName => $value,
+        ], [
+            $fieldName => $rules,
+        ]);
+
+        return $validator->fails()
+            ? $validator->errors()->first($fieldName)
+            : null;
+    }
+
+    /**
+     * Returns array of [organizationId => organizationIdentifier].
+     *
+     * @param $aidstreamOrganizationIds
+     *
+     * @return array
+     */
+    private function getAidstreamOrganizationIdentifier($aidstreamOrganizationIds): array
```

Comment:
if registry information does not exist here, we may need to take publisher_id from organizations table 'user_identifier'